pthread_util.c

```diff
@@ -39,14 +54,30 @@ int create_thread(pthread_t *thread,  void *(*start_routine) (void *), void *arg
 	if (detached) {
 		pthread_attr_init(&attr);
 		pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_DETACHED);
+/* On MacOS X we can only set a threads name from inside that thread.
+Because of this, if we're compiling for OS X, we have pthread_create
+specify a trampoline function that calls the actual routine that the 
+thread will be running. */
+#ifndef __APPLE__
 		rc = pthread_create(thread, &attr, start_routine, arg);
+#else
+		//FIXME: This could be a potential memory leak. Not  sure if
+		//we can free t_args immediately without messing up threading.
```

Comment:
Can we not free(t->args) at the end of pthread_osx_trampoline()?

I'm not seeing a reason why not, but... maybe I'm just missing it.

Also, git diff | ./checkpatch.pl - will probably complain about those C++ style comments (not that I would reject a patch like this because of that, lol.)

Reply:
I can remove those.